src/components/flash/flash.js

```diff
@@ -305,19 +318,19 @@ class Flash extends React.Component {
           key={ title }
           title={ title }
           open={ this.state.dialogs[title] || false }
-          onCancel={ this.toggleDialog.bind(this, title) }
+          onCancel={ this.toggleDialog(title) }
         >
           { desc }
         </Alert>
       );
 
       // create text for item
-      text = (
+      return (
```

Comment:
A variable to hold `text` ought to be initialised outside of the `if` then augmented by the `if` if the condition is true?

Something like:

```
var x = '';
if (y === 137) {
  x = 'yep'
}
return x;
```